src/renderer/src/components/CreateProjectModal/index.tsx

```diff
@@ -0,0 +1,286 @@
+import { useState } from 'react'
+import {
+  Container,
+  Dialog,
+  FormControlLabel,
+  FormHelperText,
+  InputLabel,
+  Radio,
+  RadioGroup,
+  TextField,
+  Typography,
+  useTheme,
+} from '@mui/material'
+import ErrorIcon from '@mui/icons-material/Error'
+import { Column, Row } from '../LayoutComponents'
+import { defineMessages, useIntl } from 'react-intl'
+import { Button } from '../Button'
+import { FormLabel } from '../FormLabel'
+import { useNavigate } from 'react-router-dom'
+
+const PROJECT_NAME_MAX_LENGTH = 100
+const DEVICE_NAME_MAX_LENGTH = 60
+
+export const CreateProjectModal = ({ open, onClose }: CreateProjectModalProps) => {
+  const intl = useIntl()
+  const theme = useTheme()
+  const navigate = useNavigate()
+  const [projectName, setProjectName] = useState('')
+  const [deviceName, setDeviceName] = useState('')
+  const [projectNameError, setProjectNameError] = useState('')
+  const [deviceNameError, setDeviceNameError] = useState('')
+  const [teamMembersShouldBeVisible, setTeamMembersShouldBeVisible] = useState<boolean | undefined>()
+  const [teamMembersObservationsEditable, setTeamMembersObservationsEditable] = useState<
+    boolean | undefined
+  >()
+
+  const handleTeamMembersVisibleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTeamMembersShouldBeVisible((event.target as HTMLInputElement).value === 'yes' ? true : false)
+  }
+  const handleObservationsEditableChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTeamMembersObservationsEditable((event.target as HTMLInputElement).value === 'yes' ? true : false)
+  }
+
+  const handleCloseDialog = (event?: Record<string, never>, reason?: 'escapeKeyDown' | 'backdropClick') => {
+    if (reason === 'backdropClick' || reason === 'escapeKeyDown') return
+    onClose()
+  }
+
+  const handleClickCreateProject = () => {
+    if (!projectName) {
+      setProjectNameError(intl.formatMessage(messages.projectNameRequired))
+    }
+
+    if (!deviceName) {
+      setDeviceNameError(intl.formatMessage(messages.deviceNameRequired))
+    }
+
+    if (projectName && deviceName) {
+      navigate('/') // Should navigate to app in observation mode, but not available in this branch yet.
+    }
+  }
```

Comment:
if `teamMembersShouldBeVisible` and `teamMembersObservationsEditable` is required, these also need to be checked in this function